manualintegrate: Add erf to LIATE rule in integration by parts

salastro opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->

Combined with #27839, solves one of the integrals in #27586. To be extended.

#### Brief description of what is fixed or changed

Placed `erf(x)` in the LIATE hierarchy after inverse trigonometric functions and before algebraic functions. The error function differentiates to an exponential, making it behave similarly to inverse trigonometric functions, which differentiate to algebraic fractions. Since integration by parts works best when `u` simplifies upon differentiation, `erf(x)` should be chosen for `u` before algebraic functions but after inverse trig functions.

#### Other comments
N/A

#### Release Notes

<!-- Write the release notes for this release below between the BEGIN and END
statements. The basic format is a bulleted list with the name of the subpackage
and the release note for this PR. For example:

* solvers
  * Added a new solver for logarithmic equations.

* functions
  * Fixed a bug with log of integers. Formerly, `log(-x)` incorrectly gave `-log(x)`.

* physics.units
  * Corrected a semantical error in the conversion between volt and statvolt which
    reported the volt as being larger than the statvolt.

or if no release note(s) should be included use:

NO ENTRY

See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more
information on how to write release notes. The bot will check your release
notes automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
* integrals
  * Some integrals involving `erf(x)` are now handled by manualintegrate.
<!-- END RELEASE NOTES -->

salastro committed:
manualintegrate: Add erf to LIATE rule in integration by parts

Placed `erf(x)` in the LIATE hierarchy after inverse trigonometric functions
and before algebraic functions. The error function differentiates to an
exponential, making it behave similarly to inverse trigonometric functions,
which differentiate to algebraic fractions. Since integration by parts works
best when `u` simplifies upon differentiation, `erf(x)` should be chosen for
`u` before algebraic functions but after inverse trig functions.

salastro commented:
Since #27839 got merged, I think this should get a look.

oscarbenjamin commented:
There should be a test.

oscarbenjamin commented:
> Is there a way to fix the auto detection of the integration method?

You would have to follow through in a debugger to see what is happening.

salastro commented:
Okay sorry for the false alert. Turns out it is because I hadn't merged the powsimp branch locally.

salastro committed:
Add a test for erf(a*x)*exp(b*x) integral

The test targets the addition of the special function erf to the LIATE
rule in manualintegrate.

salastro commented:
I think that should be good.

oscarbenjamin commented:
Looks good. I edited the release note.
